### ISSUE TITLE
rusty-hook auto hook setup with v.0.5.1

### ISSUE THREAD

**Issue by calebcartwright** [brace-rs/brace#26]
Greetings, and apologies in advance for the random ping 😄 

I noticed you were using [rusty-hook](https://crates.io/crates/rusty-hook) through a GH search and just wanted to share that we've published a new version of `rusty-hook` (0.5.1). This version enables automatically adding all the git hooks on build which eliminates the need to manually run the `init` command. 

With this new version, `rusty-hook` can simply be added as a dev-dependency 
```toml
[dev-dependencies]
rusty-hook = "0.5.1"
```

There's still a need to do a one-time `cargo install rusty-hook` on the system, but hopefully [cargo will add support to make dev-deps runnable](https://github.com/rust-lang/cargo/issues/2267) in the future and we can remove that one time install step too. 

Thanks for giving `rusty-hook` a try, and if you have any feedback/suggestions/complaints/etc. please let us know!

**Comment by dbalcomb** [comment]
Apologies @calebcartwright for not responding to this issue sooner. I appreciate the work that you are doing on rusty-hook. I had been meaning to lay out my thoughts but I have been busy over the past month or so, hence my lack of updates on this project.

My first note is that this is only a side project to help me learn Rust and I didn't think anyone would stumble upon it just yet. I have some further work to update the development workflow but as the sole person working on this I haven't had a need to make any changes as everything is working well. This includes rusty-hook which is fantastic.

The second note is that while I appreciate that rusty-hook can be added as a dev dependency I feel that it doesn't suit the needs of my project. Some of my tests are reliant on setting up a local development environment (database etc.) and I feel that requiring that setup would prevent others from contributing small updates to the project. Not that I expect that to happen but perhaps in the future. Those people could simply rely on the build pipeline to pick up any changes that need to be made. Anyone more involved could then manually initialise rusty-hook to run the required checks and tests locally.

Perhaps I could add a build task to run a subset of tests that don't require a local environment setup and get rusty-hook to use that but then as far as i know there would not be an easy way for me personally, or anyone else opting in for that matter, to automatically run all the tests. Maybe if there was a way to specify an optional local gitignored override file that could be committed to the project and copied/renamed to activate. Then maybe I could so it working. But to be honest I am happy with how it works at the moment.

My only frustration was that when I first set it up the git hooks didn't have execute permissions and SourceTree, my git client, just ignored them without warning so it wasn't without issues. Also for whatever reason SourceTree doesn't like commit hooks unless started from the terminal so it took me a few tries but everything is fine now. I notice that there is still an outstanding issue to add that functionality.

So to sum it up it sounds like a useful feature for smaller more self-contained projects but not necessarily for what I am trying to do.

**Comment by calebcartwright** [comment]
No worries! To be honest we were just surprised to see someone using it, and wanted to drop a note since it was still in a pretty raw form. I wasn't expecting any response, but thank you so much for taking the time to provide such a thoughtful one! 

This issue is admittedly way off topic for your repo (again my apologies), but I did just want to note that the exec permission issue you experienced was indeed a bug on our end that we've subsequently fixed. Everything should now just "work" (we've tested on Ubuntu, OSX, and Windows)

If you get a chance and are willing, I believe the below steps should fix everything on your system:
1. `cargo install --force rusty-hook` (since you probably have the older/buggy version installed globally on your system)
2. Follow the steps outlined in our [quick start](https://github.com/swellaby/rusty-hook#quick-start)

No worries though if you don't want to try, but either way thanks for giving it a whirl and good luck with your project!
